packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -594,6 +594,29 @@ test.describe('Invalidation', () => {
 		await page.locator('button').click();
 		await expect(page.getByText('updated')).toBeVisible();
 	});
+
+	test('goto after invalidation does not reset state', async ({ page }) => {
+		await page.goto('/load/invalidation/invalidate-then-goto');
+		const layout = await page.textContent('p.layout');
+		const _page = await page.textContent('p.page');
+		expect(layout).toBeDefined();
+		expect(_page).toBeDefined();
+
+		await page.click('button.invalidate');
+		await page.evaluate(() => window.promise);
+		const next_layout_1 = await page.textContent('p.layout');
+		const next_page_1 = await page.textContent('p.page');
+		expect(next_layout_1).not.toBe(layout);
+		expect(next_page_1).toBe(_page);
```

Comment:
I think the Playwright way to write this would be something like below where it uses its auto waiting functionality to wait until the conditions are true. I have to run, so didn't have time to check if these are the APIs with that auto waiting built-in (some of the APIs have it and some don't as I recall)
```suggestion
		await page.click('button.invalidate');
		expect(page.textContent('p.layout')).not.toBe(layout);
		expect(page.textContent('p.page')).toBe(_page);
```

Reply:
https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-text would be the assertion to probably use here for auto-waiting.

Reply:
Still doesn't change the fact that we gotta save the text to a variable to use it in the next comparison

Reply:
But it does save you from having to do the promise dance, which would be nice to avoid

Reply:
We still need to save the string into a variable for comparison later. This would make the test read very weird (first compare text content, then save that same selector to a variable).
Also this mimicks the other tests in that suite, and frankly if this is the only thing holding up this PR then I'd rather discuss this separately and merge this PR because it's an important bug to fix.